File: service/GoodNight.Service.Domain/Model/Read/Transfer/Log.cs
using System.Collections.Immutable;

namespace GoodNight.Service.Domain.Model.Read.Transfer
{
  public record Choice(
    string Kind,
    string? Text,
    string? Icon,
    IImmutableList<Property>? Effects);

  public record Log(
    uint Number,
    string Text,
    IImmutableList<Property> Effects,
    Choice Chosen);
}

File: client/src/state/model/read/Log.ts
import type {Property} from "./Property";


interface ChoiceAction {
  kind: "action";
  text: string;
  icon: string | null;
  effects: Array<Property>;
}

interface ChoiceReturn {
  kind: "return";
}

interface ChoiceContinue {
  kind: "continue";
}


export type Choice =
    | ChoiceAction
    | ChoiceReturn
    | ChoiceContinue;


export interface Log {
  number: number;
  text: string;
  effects: Array<Property>;
  chosen: Choice;
}
